src/pocketmine/Player.php

```diff
@@ -998,7 +998,12 @@ protected function doFirstSpawn(){
 		}
 
 		$this->teleport($pos);
-
+		
+		$this->setAllowFlight($this->gamemode == 3 || $this->gamemode == 1);
```

Comment:
I don't like that method name, it's misleading. I tend to use getGamemode() & 0x01